test/modules/obsagent-client/src/otel_client.rs

```diff
@@ -0,0 +1,140 @@
+use std::{
+    sync::{
+        atomic::{AtomicBool, Ordering},
+        Arc, Mutex,
+    },
+    time::Duration,
+};
+
+use futures::{Stream, StreamExt};
+use opentelemetry::{
+    global,
+    metrics::{self, ObserverResult},
+    sdk::metrics::{selectors, PushController},
+};
+use opentelemetry_otlp::ExporterConfig;
+use rand::random;
+use thiserror::Error;
+use tokio::time;
+use tracing::error;
+
+use crate::config::Config;
+
+const METER_NAME: &str = "microsoft.com/azureiot-edge";
+
+#[derive(Debug, Error)]
+pub enum OTelClientError {
+    #[error("could not initialize opentelemetry metrics pipeline: {0:?}")]
+    MetricsPipelineInitError(#[source] metrics::MetricsError),
+    #[error("error registering signal hook action: {0:?}")]
+    SignalHookRegisterError(#[source] std::io::Error),
+}
+
+// Skip first immediate tick from tokio, not needed for async_std.
+fn delayed_interval(duration: Duration) -> impl Stream<Item = time::Instant> {
+    opentelemetry::util::tokio_interval_stream(duration).skip(1)
+}
+
+fn init_meter(period: f64, otlp_endpoint: String) -> metrics::Result<PushController> {
+    let export_config = ExporterConfig {
+        endpoint: otlp_endpoint,
+        ..ExporterConfig::default()
```

Comment:
Are the `..`ExporterConfig::default() intentional? (Note: double `.`)

Reply:
This is rust syntax for "take the default values for rest of the variables"